assets/src/block-editor/index.js

```diff
@@ -52,14 +55,24 @@ const blocks = require.context( './blocks', true, /(?<!test\/)index\.js$/ );
 blocks.keys().forEach( ( modulePath ) => {
 	const { name, settings } = blocks( modulePath );
 
-	// Prevent registering latest-stories block if not enabled.
-	if ( 'amp/amp-latest-stories' === name && typeof ampLatestStoriesBlockData === 'undefined' ) {
-		return;
-	}
+	const isLatestStoriesBlock = 'amp/amp-latest-stories' === name;
 
-	const blockRequiresAmp = AMP_DEPENDENT_BLOCKS.includes( name );
+	const shouldRegister = (
+		(
+			select( 'amp/block-editor' ).isWebsiteEnabled() &&
+			(
+				( ! isLatestStoriesBlock && select( 'amp/block-editor' ).isStandardMode() && AMP_DEPENDENT_BLOCKS.includes( name ) ) ||
+				( isLatestStoriesBlock && select( 'amp/block-editor' ).isStoriesEnabled() )
```

Comment:
We already have the exact same check a few lines down, so this one here seems redundant.

Reply:
Simplified it a bit now.